include/mitsuba/render/optix/shapes.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#ifdef __CUDACC__
+// List all shape's CUDA header files to be included in the PTX code generation.
+// Those header files are located in '/mitsuba/src/shapes/optix/'
+#include "cylinder.cuh"
+#include "disk.cuh"
+#include "mesh.cuh"
+#include "rectangle.cuh"
+#include "sphere.cuh"
+#else
+NAMESPACE_BEGIN(mitsuba)
+/// List of custom OptiX shape desciptors
+/// (ClassName, ClosesthitFuncName, IntersectionFuncName)
+static std::tuple<const char*, const char*, const char*> custom_optix_shapes[] = {
```

Comment:
Taking another look at this, it seems like a very "hardcoded" way to do string concatenation :-). I would propose that we allow the shape names to be upper case in those method names, and then we can just use the normal "+" operator of `std::string` to add the prefixes.

Reply:
Actually I originally tried to concatenate those strings as you are suggesting. Somehow the optix routine that is using them (`optixProgramGroupCreate`) kept segfaulting when forging one of those string manually. I assumed it requires those `const char *` to be in the data segment somehow. (really not sure about this)